File: Java/app/src/main/java/com/tutorial/notesapp/activity/BaseActivity.java
package com.tutorial.notesapp.activity;

import android.content.ComponentCallbacks2;
import android.content.Context;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onTrimMemory(int level) {
        super.onTrimMemory(level);
        if (level <= ComponentCallbacks2.TRIM_MEMORY_BACKGROUND)
            System.gc();
    }

    public void showKeyboard(View view) {
        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.showSoftInput(view, 0);
    }

    public void hideKeyboard() {
        if (getCurrentFocus() == null) return;
        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
    }

    public void hideKeyboard(View view) {
        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }
}

File: Kotlin/app/src/main/java/com/tutorial/notesapp/activity/BaseActivity.kt
package com.tutorial.notesapp.activity

import androidx.appcompat.app.AppCompatActivity
import com.tutorial.notesapp.manager.DataManger

abstract class BaseActivity: AppCompatActivity() {

    override fun onTrimMemory(level: Int) {
        super.onTrimMemory(level)
        if (level <= TRIM_MEMORY_BACKGROUND)
            System.gc()
    }

    override fun onStop() {
        super.onStop()
        DataManger.instance.closeDatabase()
    }

    override fun onDestroy() {
        super.onDestroy()
        DataManger.instance.closeDatabase()
    }
}